pyhf/tensor/backend.py

```diff
@@ -0,0 +1,20 @@
+import inspect
+from six import string_types
+
+
+class Backend(object):
+    """General backend class for all of our backends"""
+
+    def __eq__(self, other):
+        if isinstance(other, Backend):
+            # if this is comparing to Backend objects
+            return self.__class__ == other.__class__
+        elif inspect.isclass(other):
+            # if this is comparing a Backend class
+            return self.__class__ == other
+        elif isinstance(other, string_types):
+            # if this is comparing to a string like 'numpy'
+            return self.name == other
+        else:
+            return False
+        return False # always return false
```

Comment:
do we need this base class.. as far as I can see, we only ever do use `__eq__` to compare `self.tensorlib != tensorlib` which could just be done via `self.tensorlib.name ! = tensorlib.name` 

we might very well need a base class in the future, but perhaps for now it's premature?

Reply:
We can copy/paste this code into all backends -- but it'll be a pain to maintain?

Reply:
this can be removed now that we have `.name`

Reply:
Do you not want to do somthing like `tensorlib == 'numpy'` or you want to do `tensorlib.name == 'numpy'` only?

Reply:
yeah I think attribute access is sufficient, so we don't really need the `__eq__`